### ISSUE TITLE
Warn on missing opening brace

### ISSUE THREAD

**Issue by oliv3r** [koalaman/shellcheck#1890]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related
The only thing remotely related would be SC1056

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
echo "$0}"
```

#### Here's what shellcheck currently says:
Everything is spiffy

#### Here's what I wanted or expected to see:
While I understand it's probably very hard to detect a missing opening brace, a missing closing brace IS detected. 
```sh
#!/bin/sh
echo "${0"
```
Both are equally? right or wrong. I do see that the `${` has no meaning, but `$0}` certainly does, it's just weird and unexpected. What if the user REALLY intended to print a closing brace in his string without an accompanying opening brace? I think this would be _so_ exceptional, that I do think it's fair that such a user should add `# shellcheck disable=SC1056  # Yes, I really meant to not print an opening brace here.`
Of course the proper way to write this, to avoid any shellcheck warning could be `"${0}}"` which _still_ reads weirdly and awkwardly, but at least this syntax form should work (though the earlier exception would still seem sensible even here).
